### PR TITLE
Integrate Merkle Blob into Data Layer

### PR DESCRIPTION
**Pull request by fchirica**

### CODE REVIEW

File: chia/data_layer/data_store.py [hunk 1]
```diff
@@ -2065,94 +1489,38 @@ async def remove_subscriptions(self, store_id: bytes32, urls: list[str]) -> None
...
-                    if ref_counts[right] == 0:
-                        queue.append(right)
-
     async def unsubscribe(self, store_id: bytes32) -> None:
         async with self.db_wrapper.writer() as writer:
             await writer.execute(
                 "DELETE FROM subscriptions WHERE tree_id == :tree_id",
                 {"tree_id": store_id},
             )
+            await writer.execute(
```
Comment (altendky) [inline chia/data_layer/data_store.py:1498]: I thought we intentionally had separation of unsubscribing and deleting data.

File: chia/_tests/core/data_layer/test_data_store.py [hunk 2]
```diff
@@ -1127,28 +961,27 @@ async def test_kv_diff(data_store: DataStore, store_id: bytes32) -> None:
...
         value = (i + 200).to_bytes(4, byteorder="big")
-        seed = leaf_hash(key=key, value=value)
-        node_hash = await data_store.get_terminal_node_for_seed(store_id, seed)
+
         if random.randint(0, 4) > 0 or insertions < 10:
             insertions += 1
-            side = None if node_hash is None else data_store.get_side_for_seed(seed)
-
-            await data_store.insert(
+            await data_store.autoinsert(
```
Comment (altendky) [inline chia/_tests/core/data_layer/test_data_store.py:972]: Does this test pattern need to change or can it stay the same?  Or maybe there's another reason to change it?

File: chia/_tests/core/data_layer/test_data_store.py [hunk 3]
```diff
@@ -1854,30 +1553,34 @@ async def test_insert_from_delta_file(
             store_id=store_id,
             status=Status.COMMITTED,
         )
+        await data_store.add_node_hashes(store_id)
 
     root = await data_store.get_tree_root(store_id=store_id)
-    assert root.generation == num_files + 1
+    assert root.generation == num_files
```
Comment (altendky) [inline chia/_tests/core/data_layer/test_data_store.py:1559]: It seems like changes to this ought to be separate from changing out the data store backend?

**Review by altendky — CHANGES_REQUESTED** [review]

File: chia/_tests/core/data_layer/test_data_store.py (hunk 3 above)
Comment (fchirica) [reply chia/_tests/core/data_layer/test_data_store.py:1559]: I think it was off by one before (we did an unnecessary `create_tree`, introducing another empty root at generation=1)

File: chia/data_layer/data_store.py (hunk 1 above)
Comment (fchirica) [reply chia/data_layer/data_store.py:1498]: as far as I can remember, it wasn't intentional - it was a way to make delete data non blocking of other operations. but duplicating everything by store_id in this new schema solves this (we're deleting fast, as opposed to before where we'd waste time and block everything)

File: chia/data_layer/data_store.py [hunk 4]
```diff
@@ -184,20 +187,85 @@ async def transaction(self) -> AsyncIterator[None]:
...
             cursor = await reader.execute("SELECT * FROM schema")
-            row = await cursor.fetchone()
-            if row is not None:
+            rows = await cursor.fetchall()
+            all_versions = {"v1.0", "v2.0"}
+
+            for row in rows:
                 version = row["version_id"]
-                if version != "v1.0":
+                if version not in all_versions:
```
[the remark this answers is not in the document]
Comment (fchirica) [reply chia/data_layer/data_store.py:247]: the previous migration would create tables that we'd immediately have to delete in the next migration. that's why I thought to 'shortcut' it here.

[the remark this answers is not in the document]
Comment (fchirica) [reply chia/data_layer/data_store.py:247]: I didn't get to test pre-v1.0 to v2.0 tho (just 1.0 to 2.0), but in principle it should work the same.

File: chia/data_layer/data_store.py (hunk 1 above)
Comment (altendky) [reply chia/data_layer/data_store.py:1498]: maybe i'm thinking of the `.dat` deletion that's controllable via the cli `--retain` option.  alrighty.

**Review by altendky — APPROVED** [review]

**Pull request merged**